src/types.ts

```diff
@@ -97,13 +97,13 @@ export interface Ctx {
   _random?: {
     seed: string | number;
   };
+  random: RandomAPI;
 
   // TODO public api should have these as non-optional
   // internally there are two contexts, one is a serialized POJO and another
   // "enhanced" context that has plugin api methods attached
   events?: EventsAPI;
   log?: LogAPI;
-  random?: RandomAPI;
```

Comment:
This should also apply to the other default plugins `log` and `events` as they suffer from the same problem and will also be always present in moves etc. but not on the client.